diff -rupN --no-dereference binutils-2.41/config/override.m4 binutils-2.41-new/config/override.m4
--- binutils-2.41/config/override.m4	2023-07-03 01:00:00.000000000 +0200
+++ binutils-2.41-new/config/override.m4	2023-10-17 08:36:14.285923364 +0200
@@ -41,7 +41,7 @@ dnl Or for updating the whole tree at on
 AC_DEFUN([_GCC_AUTOCONF_VERSION_CHECK],
 [m4_if(m4_defn([_GCC_AUTOCONF_VERSION]),
   m4_defn([m4_PACKAGE_VERSION]), [],
-  [m4_fatal([Please use exactly Autoconf ]_GCC_AUTOCONF_VERSION[ instead of ]m4_defn([m4_PACKAGE_VERSION])[.])])
+  [])
 ])
 m4_define([AC_INIT], m4_defn([AC_INIT])[
 _GCC_AUTOCONF_VERSION_CHECK
